File: app/src/main/java/prototypez/github/io/sq/demo/fragment/LoginBySmsFragment.java
package prototypez.github.io.sq.demo.fragment;

import android.os.Bundle;
import android.support.annotation.NonNull;
import android.support.annotation.Nullable;
import android.support.v4.app.Fragment;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.TextView;

import com.jyn.vcview.VerificationCodeView;

import prototypez.github.io.sq.demo.R;
import prototypez.github.io.sq.demo.entity.User;
import prototypez.github.io.sq.util.BundleBuilder;

public class LoginBySmsFragment extends Fragment {

    VerificationCodeView mVerificationCodeView;

    TextView mHint;

    LoginBySmsCallback mCallback;

    User mUser;

    public void setParams(User user) {
        setArguments(BundleBuilder.newInstance().putSerializable("user", user).build());
    }

    @Override
    public void onCreate(@Nullable Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        mUser = (User) getArguments().getSerializable("user");
    }

    @Nullable
    @Override
    public View onCreateView(@NonNull LayoutInflater inflater, @Nullable ViewGroup container, @Nullable Bundle savedInstanceState) {
        View view = inflater.inflate(R.layout.fragment_login_by_sms, container, false);

        mVerificationCodeView = view.findViewById(R.id.code);
        mHint = view.findViewById(R.id.hint);

        mHint.setText(String.format("验证码已发送至: %s", mUser.phone));

        mVerificationCodeView.setOnCodeFinishListener(new VerificationCodeView.OnCodeFinishListener() {
            @Override
            public void onComplete(String content) {
                // mock success
                if (mCallback != null) {
                    mCallback.onLoginOk(new User(
                            mUser.id,
                            mUser.nickName,
                            mUser.phone,
                            "Shanghai",
                            29
                    ));
                }
            }
        });
        return view;

    }

    public void setLoginBySmsCallback(LoginBySmsCallback callback) {
        this.mCallback = callback;
    }


    public interface LoginBySmsCallback {
        void onLoginOk(User user);
    }
}

File: app/src/main/java/prototypez/github/io/sq/demo/processes/login/steps/LoginBySmsFragment.kt
package prototypez.github.io.sq.demo.processes.login.steps
import kotlinx.android.synthetic.main.fragment_sms_verify.*

import android.os.Bundle
import android.support.v4.app.Fragment
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup

import com.jyn.vcview.VerificationCodeView

import prototypez.github.io.sq.demo.R
import prototypez.github.io.sq.demo.entity.User
import prototypez.github.io.sq.util.BundleBuilder

class LoginBySmsFragment : Fragment() {

    lateinit var loginBySmsCallback: (User) -> Unit

    private lateinit var mUser: User

    fun setParams(user: User) {
        arguments = BundleBuilder.newInstance().putSerializable("user", user).build()
    }

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        mUser = arguments!!.getSerializable("user") as User
    }

    override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?, savedInstanceState: Bundle?): View? {
        return inflater.inflate(R.layout.fragment_sms_verify, container, false)
    }

    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
        super.onViewCreated(view, savedInstanceState)
        tv_hint.text = String.format("验证码已发送至: %s", mUser.phone)

        vc_code.onCodeFinishListener = VerificationCodeView.OnCodeFinishListener {
            // mock success
            loginBySmsCallback(User(
                    mUser.id,
                    mUser.nickName,
                    mUser.phone,
                    "Shanghai",
                    29,
                    false
            ))
        }
    }
}
